syntax = "proto3";

package sr.grpc;
option java_multiple_files = true;

message MeetingRequest {
    string team = 1;
}

message Person {
    string name = 1;
    string surname = 2;
}

enum Type {
    STANDUP = 0;
    COMPANY_WIDE = 1;
    DIRECTORS = 2;
}

message MeetingResponse {
    string team = 1;
    string room = 2;
    string datetime = 3;
    Type type = 5;
    repeated Person participants = 4;
}

enum Conditions {
    SUNNY = 0;
    CLOUDY = 1;
    RAINY = 2;
    STORMY = 3;
}

message WeatherRequest {
    string city = 1;
    float desiredTemperature = 2;
    float tolerance = 3;
}

message WeatherResponse {
    string city = 1;
    float temperature = 2;
    Conditions conditions = 3;
}

service SubscriptionService {
rpc subscribeWeather(WeatherRequest) returns (stream WeatherResponse);
rpc subscribeMeetings(MeetingRequest) returns (stream MeetingResponse);
}

// D:\Dokumenty\rozprochy\protoc-3.11.4-win64\bin\protoc.exe -I=. --java_out=gen --plugin=protoc-gen-grpc-java=D:\Dokumenty\rozprochy\protoc-3.11.4-win64\bin\protoc-gen-grpc-java.exe --grpc-java_out=./gen subscriptions.proto
// python -m grpc_tools.protoc -I . --python_out=./gen --grpc_python_out=./gen subscriptions.proto